src/lib/dialog/dialog.ts

```diff
@@ -39,21 +39,21 @@ export class MdDialog {
    * @param component Type of the component to load into the load.
    * @param config
    */
-  open<T>(component: ComponentType<T>, config: MdDialogConfig): Promise<MdDialogRef<T>> {
-    let overlayRef: OverlayRef;
+  open<T>(component: ComponentType<T>, config: MdDialogConfig): MdDialogRef<T> {
+    let overlayRef = this._createOverlay(config);
+    let dialogContainer = this._attachDialogContainer(overlayRef, config);
 
-    return this._createOverlay(config)
-        .then(overlay => overlayRef = overlay)
-        .then(overlay => this._attachDialogContainer(overlay, config))
-        .then(containerRef => this._attachDialogContent(component, containerRef, overlayRef));
+    // TODO: probably need to wait for dialogContainer ngOnInit before attaching.
```

Comment:
Was an in-progress thought while I was working, ended up not being the case. Removed.